runtime/src/chpl-gpu.c

```diff
@@ -41,6 +41,23 @@ void chpl_gpu_init(void) {
 
 void chpl_gpu_on_std_modules_finished_initializing(void) {
   chpl_gpu_impl_on_std_modules_finished_initializing();
+
+  // this function is something that we're not proud of already. The following
```

Comment:
Yeah I noticed the behavior you noted in this comment as well (and would actually go as far as temporarily changing CHPL_GPU_DEBUG to printfs just so I could see the output).

Sometimes I like having that extra verification that we're using the `array_on_device` strategy so I appreciate you found a way to get that output to print even if it's in kind of a strange place.